src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
```

Comment:
[CODE PERFOMANCE] - You have an unnecessary method `checkHealth` that you call after each `bite`. This causes an additional iteration over the `Animal.alive` array each time you call `bite`. You can avoid this by moving the filter operation into the `bite` method right after decreasing the herbivore's health. This way you only have to iterate over the array once, when the herbivore's health becomes less than or equal to 0.